internal/rpcserver/interface.go

```diff
@@ -516,3 +505,33 @@ type BlockTemplater interface {
 	// time while taking into account the consensus rules.
 	UpdateBlockTime(header *wire.BlockHeader) error
 }
+
+// Filterer provides an interface for retrieving a block's committed filter or
+// committed filter header.
+//
+// The interface contract requires that all of these methods are safe for
+// concurrent access.
+type Filterer interface {
+	// FilterByBlockHash returns the serialized contents of a block's regular or
+	// extended committed filter.
+	FilterByBlockHash(h *chainhash.Hash, filterType wire.FilterType) ([]byte, error)
+
+	// FilterHeaderByBlockHash returns the serialized contents of a block's regular
+	// or extended committed filter header.
+	FilterHeaderByBlockHash(h *chainhash.Hash, filterType wire.FilterType) ([]byte, error)
+}
+
+// FiltererV2 provides an interface for retrieving a block's version 2 GCS
+// filter.
+//
+// The interface contract requires that all of these methods are safe for
+// concurrent access.
+type FiltererV2 interface {
+	// FilterByBlockHash returns the version 2 GCS filter for the given block hash
+	// when it exists.  This function returns the filters regardless of whether or
+	// not their associated block is part of the main chain.
+	//
+	// An error of type NoFilterError must be returned when the filter for the
```

Comment:
This probably should call out `blockchain.NoFilterError` since otherwise it could be ambiguous.

Reply:
Good call, updated.